### DISCUSSION TITLE
Online help

### DISCUSSION THREAD

**Discussion by atalayismail** [gijzelaerr/python-snap7#249] [Q&A] [answered]
Can someone give me an online help for the installing this library  ?

**Comment by swamper123** [comment]
Have you done, what I told you in #244?

**Reply by atalayismail** [reply]
Yes i tried three times somethings are not proper on this method i think can you help me ?

**Reply by swamper123** [reply]
Just to reproduce the steps again, you are using a Windows device. You copied all files either from Win32, Win64 or Legacy (depending on your PC) into a path from PATH (the Window "Umgebungsvariablen" and there in the Subwindow(?) "Systemvariablen") you can find here:

![grafik](https://user-images.githubusercontent.com/24408657/107473547-2b7cae80-6b71-11eb-90ab-90514a8e1ba1.png)

You checked, that you are using the right python enviroment (if you use a venv, you have to activate it first, before installing python-snap7 via pip).

And you tried to just import snap7.

If you do all the things up here, please provide us with more information. What is happening? What errors occure?

**Reply by swamper123** [reply]
Additionally: Check if your Python Version is equivalent to your the Windows version (so if python is 32bit, you have to use the files from win32 as well, dito python 64 bit needs win64 files).

**Comment by atalayismail** [comment]
where am i missing i added all of the images and the error result what is your advice ? 
![error1](https://user-images.githubusercontent.com/48536582/107502880-4f101b00-6baa-11eb-8d0a-47c922b9b72b.JPG)
![error2](https://user-images.githubusercontent.com/48536582/107502882-50414800-6baa-11eb-8203-cde9fbe3e3e7.JPG)
![error3](https://user-images.githubusercontent.com/48536582/107502883-50414800-6baa-11eb-9346-f6f045c37fda.JPG)
![error4](https://user-images.githubusercontent.com/48536582/107502884-50d9de80-6baa-11eb-8592-d28ceb1774ef.JPG)

**Comment by swamper123** [comment]
What .dll have you chose (Win32, Win64 or Legacy)?
I guess you have a 64bit system running?
What does your Python Interpreter return if you execute: 
```python
import sys

def is_64bit() -> bool:
    return sys.maxsize > 2**32

```
Since you are using Pycharm, do you have created a virutal enviroment for your project or not (Pycharm usually asks if it should create one, when you create a new Project)?

If you have created an virtual enviroment, your Path for the dlls will be wrong. I would prefere then to throw them into the _Windows/system32_ folder or create another folder and add that one to Path, like you did with D:\Python.

**Reply by swamper123** [reply]
Any progress?

**Reply by atalayismail** [reply]
No, i think i am not able to do it you can close the topics.

**Reply by lautarodapin** [reply]
You could edit the python-snap7 lib so you can, specify where to find the .dll

**Comment by gijzelaerr** [comment, marked as answer]
Closing this as the original poster requested. Additionally, python-snap7 3.0 is now a pure Python implementation and no longer requires any native .dll/.so files, so the installation issues described here no longer apply.
